tests/tests/test_compat.py

```diff
@@ -34,56 +34,69 @@
 container_factory = factory.get_factory()
 
 TIMEOUT = timedelta(minutes=5)
+"""
+ COMPAT_MENDER_VERSIONS array stores the versions of the virtual device images we test against.
+ In order to add a new client: add a new a composition file in
+extra/integration-testing/test-compat/docker-compose.compat-VERSION.yml
+where VERSION is the version string.
+"""
+COMPAT_MENDER_VERSIONS = factory.DockerComposeCompatibilitySetup.get_versions()
 
 
 @pytest.fixture(scope="function")
-def setup_os_compat(request):
-    env = container_factory.get_compatibility_setup()
-    request.addfinalizer(env.teardown)
-    env.setup()
+def setup_os_compat():
+    def create(tag):
+        env = container_factory.get_compatibility_setup(tag=tag)
+        env.setup()
 
-    env.user = create_user(
-        "test@mender.io", "correcthorse", containers_namespace=env.name
-    )
-    env.populate_clients()
+        env.user = create_user(
+            "test@mender.io", "correcthorse", containers_namespace=env.name
+        )
+        env.populate_clients()
+
+        clients = env.get_mender_clients()
+        assert len(clients) == 1, "Failed to setup clients: expecting exactly one."
+        env.devices = []
+        for client in clients:
+            dev = MenderDevice(client)
+            dev.ssh_is_opened()
+            env.devices.append(dev)
+            break
```

Comment:
Minor: You already know that `len(clients) == 1`. Suggest removing the loop and the break and replacing with `dev = MenderDevice(clients[0])`.

Reply:
fixed.